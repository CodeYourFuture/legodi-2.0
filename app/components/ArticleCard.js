import React from 'react'
import {
  Text,
  StyleSheet,
  TouchableOpacity,
  Image,
  View
} from 'react-native'

const ArticleCard = (props) => {
  const { language } = props;
  return (
    <TouchableOpacity
      style={[styles.container, styles.text]}
      onPress={props.navigateToArticle}>
      <Image style={styles.image} source={{uri: 'http://placehold.it/100x80'}} />
      <View style={styles.content}>
        <Text style={ language === 'ar' ? styles.arabicTitle : styles.title }>{props.title}</Text>
        <Text style={ language === 'ar' ? styles.arabicDescription : styles.description } numberOfLines={2} ellipsizeMode={'tail'}>{props.description}</Text>
      </View>
    </TouchableOpacity>
  )
}

export default ArticleCard

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    paddingBottom: 10
  },
  text: {
    width: 250
  },
  content: {
    flexDirection: 'column',
    paddingLeft: 5
  },
  image: {
    width: 100,
    height: 80
  },
  title: {
    fontWeight: 'bold'
  },
  description: {
    fontSize: 15
  },
  arabicTitle: {
    textAlign: 'right',
    fontWeight: 'bold'
  },
  arabicDescription: {
    fontSize: 15,
    textAlign: 'right'
  }
})
