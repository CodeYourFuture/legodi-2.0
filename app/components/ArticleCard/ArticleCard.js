import React from 'react'
import {
  Text,
  TouchableOpacity,
  Image,
  View
} from 'react-native'
import styles from './ArticleStyle'
import PropTypes from 'prop-types'

const ArticleCard = (props) => {
  return (
    <TouchableOpacity
      style={[styles.container, styles.text]}
      onPress={props.navigateToArticle}>
      <Image style={styles.image} source={{ uri: 'http://placehold.it/100x80' }} />
      <View style={styles.content}>
        <Text style={styles.title}>{props.title}</Text>
        <Text style={styles.description} numberOfLines={2} ellipsizeMode={'tail'}>{props.description}</Text>
      </View>
    </TouchableOpacity>
  )
}

ArticleCard.propTypes = {
  navigateToArticle: PropTypes.func,
  title: PropTypes.string,
  description: PropTypes.string
}

export default ArticleCard
