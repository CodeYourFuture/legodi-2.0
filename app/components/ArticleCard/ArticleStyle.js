import {
  StyleSheet
} from 'react-native'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    paddingBottom: 10
  },
  text: {
    width: 250
  },
  content: {
    flexDirection: 'column',
    paddingLeft: 5
  },
  image: {
    width: 100,
    height: 80
  },
  title: {
    fontWeight: 'bold'
  },
  description: {
    fontSize: 15
  },
  arabicTitle: {
    fontWeight: 'bold',
    textAlign: 'right'
  },
  arabicDescription: {
    fontSize: 15,
    textAlign: 'right'
  }
})

export default styles
