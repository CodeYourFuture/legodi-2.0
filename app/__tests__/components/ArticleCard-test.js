import React from 'react'
import renderer from 'react-test-renderer'
import { Text, TouchableOpacity, Image, View } from 'react-native'
import ArticleCard from '../../components/ArticleCard/ArticleCard'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ArticleCard>
        <TouchableOpacity style={{}}>
          <Image style={{width: 100}} source={{ uri: 'http://placehold.it/100x80' }} />{' '}
          <View style={{paddingLeft: 5}}>
            <Text style={{fontWeight: 'bold'}}>Some Text</Text>
            <Text style={{fontWeight: 'bold'}}>Some Text</Text>
          </View>
        </TouchableOpacity>
      </ArticleCard>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
