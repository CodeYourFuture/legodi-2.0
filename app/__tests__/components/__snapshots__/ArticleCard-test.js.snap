// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard Compnent renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 10,
      "width": 250,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "uri": "http://placehold.it/100x80",
      }
    }
    style={
      Object {
        "height": 80,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "fontSize": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`ArticleCard Component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 10,
      "width": 250,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "uri": "http://placehold.it/100x80",
      }
    }
    style={
      Object {
        "height": 80,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "fontSize": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 10,
      "width": 250,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "uri": "http://placehold.it/100x80",
      }
    }
    style={
      Object {
        "height": 80,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "fontSize": 15,
        }
      }
    />
  </View>
</View>
`;
